ttnn/cpp/ttnn/operations/ccl/kernels/edm/erisc_async_datamover.hpp

```diff
@@ -356,8 +356,25 @@ class QueueIndexPointer {
  * of the link is ready to start sending/receiving messages. We perform a handshake to ensure that's
  * case. Before handshaking, we make sure to clear any of the channel sync datastructures local
  * to our core.
+ *
+ * The handshaking process is split into two parts for the sender/master and two parts for the
+ * the slave. The handshake is broken into 2 parts so that the master can initiate the handshake
```

Comment:
Could you please add some more detail for the the two parts involved in the handshake protocol for both sender/receiver?

Reply:
The sender would initiate the handshake and waiit for the response in the same function call (effectively blocking it from doing anything). The two parts is that now the initiate and check for response are separated. The master can now initiate the handshake right away and start messaging certain workers while it waits for a response. This can save several us